tests/Doctrine/Tests/ORM/Functional/Ticket/GH11149Test.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH11149Test extends OrmFunctionalTestCase
```

Comment:
To avoid issues with php < 8.1, you should use `@require PHP 8.1` on this class, and move at least the enum to a separate file. To keep things tidy, please put all this inside a `tests/Doctrine/Tests/ORM/Functional/Ticket/GH11149`.

Reply:
Done.